File: server/service/tello.py
from djitellopy import Tello
import time

def connect():
    time.sleep(1)
    print("connect")
    return
    tello.connect()

def demarrer():
    time.sleep(1)
    print("demarrer")
    return
    tello.takeoff()

def arreter():
    time.sleep(1)
    print("arreter")
    return
    tello.land()

def avancer(value):
    time.sleep(1)
    print("avancer")
    return
    tello.move_forward(value)

def reculer(value):
    time.sleep(1)
    print("reculer")
    return
    tello.move_back(value)

def droite(value):
    time.sleep(1)
    print("droite")
    return
    tello.move_right(value)

def gauche(value):
    time.sleep(1)
    print("gauche")
    return
    tello.move_left(value)

def haut(value):
    time.sleep(1)
    print("haut")
    return
    tello.move_up(value)

def bas(value):
    time.sleep(1)
    print("bas")
    return
    tello.move_down(value)

def emergency():
    time.sleep(1)
    print("emergency")
    return
    tello.emergency()

File: src/utils/tello.ts
import { LaunchActionType } from "../components/views/ControlPad/ControlPad";
import {
  arreter,
  avancer,
  bas,
  demarrer,
  droite,
  emergency,
  gauche,
  haut,
  reculer,
} from "../services/api/tello-api";
import { BlockName } from "../types/ScriptTypes/Block.types";

export function handleMoveTello(
  name: BlockName,
  launchAction: LaunchActionType
) {
  switch (name) {
    case BlockName.Avancer:
      launchAction(avancer);
      break;
    case BlockName.Reculer:
      launchAction(reculer);
      break;
    case BlockName.Droite:
      launchAction(droite);
      break;
    case BlockName.Gauche:
      launchAction(gauche);
      break;
    case BlockName.Demarrer:
      launchAction(demarrer);
      break;
    case BlockName.Arreter:
      launchAction(arreter);
      break;
    case BlockName.Haut:
      launchAction(haut);
      break;
    case BlockName.Bas:
      launchAction(bas);
      break;
    case BlockName.Emergency:
      launchAction(emergency);
      break;
  }
}
